simulations/vip-289/bscmainnet.ts

```diff
@@ -0,0 +1,70 @@
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { ethers } from "hardhat";
+
+import { expectEvents } from "../../src/utils";
+import { forking, testVip } from "../../src/vip-framework";
+import vip289, {
+  BTC,
+  BTC_DISTRIBUTION_SPEED,
+  ETH,
+  ETH_DISTRIBUTION_SPEED,
+  ETH_PRIME_CONVERTER,
+  PLP,
+  PSR,
+  USDC,
+  USDC_DISTRIBUTION_SPEED,
+  USDC_PRIME_CONVERTER,
+  USDT,
+  USDT_DISTRIBUTION_SPEED,
+} from "../../vips/vip-289/bscmainnet";
+import PLP_ABI from "./abi/PrimeLiquidityProvider.json";
+import PSR_ABI from "./abi/ProtocolShareReserve.json";
+
+forking(37874421, () => {
+  let psr: Contract;
+  let plp: Contract;
+  const distributionTargets: object[] = [];
+
+  before(async () => {
+    psr = await ethers.getContractAt(PSR_ABI, PSR);
+    plp = await ethers.getContractAt(PLP_ABI, PLP);
+
+    const distributionsLength = await psr.totalDistributions();
+    for (let i = 0; i < distributionsLength; i++) {
+      distributionTargets.push(await psr.distributionTargets(i));
+    }
+  });
+
+  testVip("VIP-289 Prime Adjustment", vip289(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(txResponse, [PSR_ABI], ["DistributionConfigUpdated"], [2]);
+      await expectEvents(txResponse, [PLP_ABI], ["TokenDistributionSpeedUpdated"], [4]);
+    },
+  });
+
+  describe("Post-Execution", () => {
+    it("should update the distribution config", async () => {
+      const distributionsLength = await psr.totalDistributions();
```

Comment:
I would confirm the number of distribution rules is the same before and after the VIP execution

Reply:
Done https://github.com/VenusProtocol/vips/pull/261/commits/c80b6817a03e66fdc08f37c638547d9416de3140